src/Microsoft.Extensions.Configuration.UserSecrets/ConfigurationExtensions.cs

```diff
@@ -28,11 +28,18 @@ public static IConfigurationBuilder AddUserSecrets<T>(this IConfigurationBuilder
             => configuration.AddUserSecrets(typeof(T).GetTypeInfo().Assembly);
 
         /// <summary>
+        /// <para>
+        /// This method is obsolete and will be removed in a future version.
+        /// The recommended alternative is <see cref="AddUserSecrets(IConfigurationBuilder, string)"/>. 
+        /// </para>
+        /// <para>
         /// Adds the user secrets configuration source. Searches the assembly from <see cref="Assembly.GetEntryAssembly"/>
         /// for an instance of <see cref="UserSecretsIdAttribute"/>.
+        /// </para>
         /// </summary>
         /// <param name="configuration"></param>
         /// <returns></returns>
+        [Obsolete("This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.Extensions.Configuration.ConfigurationExtensions.AddUserSecrets(this IConfigurationBuilder configuration, string userSecretsId).")]
```

Comment:
(repeating my last comment here since it seems to be hidden)
I understand I missed a conversation about this, but how are customers going to figure out how to use this overload? Do they need to go extract the user secrets Id from its present location? 
I am worried that people may need to read the manual for this then the whole migration from project.json to csproj will be harder.